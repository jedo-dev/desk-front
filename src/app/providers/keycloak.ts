import Keycloak from 'keycloak-js';
const keycloak = new Keycloak({
  url: 'http://keycloak.udptest.cloudns.nz',
  realm: 'PABP',
  clientId: 'pabp-frontend-client',
});

export default keycloak;
