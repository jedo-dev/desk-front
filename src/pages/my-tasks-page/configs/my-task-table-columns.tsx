import { Flex, TableProps } from 'antd';
import dayjs from 'dayjs';

export const columns: TableProps['columns'] = [
  {
    key: 'processName',
    align: 'center',
    width: 150,
    title: 'Процесс',
    dataIndex: 'processName',
    filterDropdown: null,
    fixed: 'left',
  },
  {
    key: 'taskName',
    align: 'center',
    width: 150,
    title: 'Дата события',
    dataIndex: 'taskName',
    fixed: 'left',
  },
  {
    key: 'startDate',
    align: 'center',
    width: 100,
    title: 'АЭС',
    render: (obj) => {
      if (obj) {
        return dayjs(obj).format('LLL');
      } else {
        return 'не указано';
      }
    },
    dataIndex: ['npp', 'code'],
  },
  {
    key: 'block.code',
    align: 'center',
    width: 100,
    title: 'Блок',
    render: (obj) => {
      if (obj) {
        if (obj.dueDate) {
          return 'Не указано';
        } else {
          return dayjs(obj.dueDate).format('LLL');
        }
      } else {
        return 'Не указано';
      }
    },
    dataIndex: '',
  },
  {
    title: 'Действия',
    dataIndex: 'id',
    key: 'lzmkabymii8e2fg1w6s',
    align: 'center',
    width: '120px',
    fixed: 'right',
    render: (_, record) => {
      return (
        <Flex gap='small'>
          <div>ffff</div>
        </Flex>
      );
    },
  },
];

export const dataDefault = {
  content: [
    // {
    //   id: 24894,
    //   taskId: '5066464',
    //   processName: 'ФПП/ФСС - 5066410',
    //   taskName: 'Редактирование черновика карточки ФПП/ФСС',
    //   startDate: 1718616728340,
    //   dueDate: 879973200000,
    //   formKey: 'PpDraft',
    //   localVariables: {
    //     equipmentOwnerLogin: null,
    //     agent: null,
    //     addReportReasons: null,
    //     actionFDocTemp: null,
    //     isPates: null,
    //     assignFDocTemp: null,
    //     dueDate: '1997-11-20T00:00:00',
    //     organizationAddReport: null,
    //     commentHistory: [
    //       {
    //         lastName: 'Админов',
    //         firstName: 'Админ Админович',
    //         data: '2024-06-17T09:32:07.877Z',
    //         action: {
    //           color: 'green',
    //           title: 'Создание черновика ',
    //           type: 'action',
    //         },
    //         taskName: 'Создание черновика проекта формуляра руководителем',
    //       },
    //     ],
    //     workerGroupLogin: null,
    //     orderExecutorLogin: null,
    //     instructionList: [
    //       {
    //         id: '893e9647a09',
    //         isTableInstruction: true,
    //         status: {
    //           id: 63,
    //           code: 'ASSIGN_NEW',
    //           status: 'Новый',
    //           docType: {
    //             id: 'ASSIGN',
    //             name: 'Поручение',
    //             viewLink: '/instructions/view/',
    //             dbTable: 'assign_f_doc',
    //           },
    //           label: 'Новый',
    //         },
    //         type: {
    //           id: 2,
    //           name: 'Стандартный',
    //           code: 'STANDARD',
    //         },
    //         dueNumber: 11,
    //         assignee: '1e3e4a12-c505-4b3a-a47d-a61ad8b72720',
    //         approver: '1e3e4a12-c505-4b3a-a47d-a61ad8b72720',
    //         body: 'Ознакомиться с результатами согласования ФПП/ФСС, оповестить ответственное за внедрение подразделение',
    //         initiator: {
    //           id: '61fa7ea7-0daa-461c-8413-cb96c77cf2d9',
    //           unsiUserId: null,
    //           firstName: 'Админ Админович',
    //           lastName: 'Админов',
    //           roles: [
    //             {
    //               id: 'f05d2f38-7e77-4ed1-ad6b-1a0e79412a79',
    //               name: 'Ответственный за отправку КП ОЭ',
    //               code: 'SENDER-KP-OE',
    //               realmId: 'PABP',
    //             },
    //             {
    //               id: 'b697c922-5a7c-4d72-ab60-2f46f6b55c3a',
    //               name: 'Ответственный за работу с карточками СНУ',
    //               code: 'SNU_RESPONSIBLE',
    //               realmId: 'PABP',
    //             },
    //             {
    //               id: '87a42af9-978f-442a-831c-be431d31ad96',
    //               name: 'Администратор ПАБП',
    //               code: 'admin',
    //               realmId: 'PABP',
    //             },
    //             {
    //               id: '79a280fe-f0a8-4bcc-b64c-71ae1a3d43c0',
    //               name: 'Базовый пользователь ПАиО',
    //               code: 'BASE_USER_PAIO',
    //               realmId: 'PABP',
    //             },
    //             {
    //               id: '3b25b467-a793-4e94-9532-4f6c877140f8',
    //               name: 'Ответственный за сбор данных',
    //               code: 'DATA_EXTRACTOR',
    //               realmId: 'PABP',
    //             },
    //             {
    //               id: '466e18dd-e459-4c76-be32-cbf0845c60c0',
    //               name: 'Ответственный ОИОЭиРН (анализ СНУ)',
    //               code: 'SNU_OIOEIRN',
    //               realmId: 'PABP',
    //             },
    //           ],
    //           subdivision: '128467',
    //           subdivisionTxt: 'Участок №1',
    //           groups: [
    //             {
    //               id: '5c0d2e87-b013-49c1-ada7-5f449566bb28',
    //               name: 'BAL',
    //             },
    //           ],
    //           email: 'admin_admin@click-group.ru',
    //           realmId: 'PABP',
    //           login: 'admin',
    //           created: '2023-02-14T12:48:02.629+00:00',
    //           password: '',
    //           active: true,
    //           isUnsi: false,
    //           position: 'Инженер',
    //           domain: {
    //             id: 1,
    //             loginPrefix: 'co',
    //             description: 'Ldap Центрального Аппарата',
    //             alias: 'Центральный аппарат',
    //             code: 'CO',
    //             federationLink: '2deebc32-7622-446a-bb0a-f4bcd26178c9',
    //           },
    //           label: 'Админов Админ Админович - Инженер - Участок №1',
    //           updatedDate: '2023-12-19T09:49:49.429+00:00',
    //         },
    //         pabpRNpp: {
    //           id: 11,
    //           createdBy: null,
    //           createdDate: '2023-03-02T13:27:25.027+00:00',
    //           updatedBy: null,
    //           updatedDate: null,
    //           shortName: 'ПАТ',
    //           fullName: 'Плавучая атомная теплоэлектростанция',
    //           code: 'PAT',
    //           priznak: 'PATES',
    //           label: 'Плавучая атомная теплоэлектростанция',
    //         },
    //         nonRemovable: true,
    //       },
    //       {
    //         id: '93e9647a095',
    //         isTableInstruction: true,
    //         status: {
    //           id: 63,
    //           code: 'ASSIGN_NEW',
    //           status: 'Новый',
    //           docType: {
    //             id: 'ASSIGN',
    //             name: 'Поручение',
    //             viewLink: '/instructions/view/',
    //             dbTable: 'assign_f_doc',
    //           },
    //           label: 'Новый',
    //         },
    //         type: {
    //           id: 2,
    //           name: 'Стандартный',
    //           code: 'STANDARD',
    //         },
    //         dueNumber: 11,
    //         assignee: '33446bd0-f369-46f3-a7f7-f2b892243c9f',
    //         approver: '33446bd0-f369-46f3-a7f7-f2b892243c9f',
    //         body: 'Ознакомиться с результатами согласования ФПП/ФСС, оповестить ответственное за внедрение подразделение',
    //         initiator: {
    //           id: '61fa7ea7-0daa-461c-8413-cb96c77cf2d9',
    //           unsiUserId: null,
    //           firstName: 'Админ Админович',
    //           lastName: 'Админов',
    //           roles: [
    //             {
    //               id: 'f05d2f38-7e77-4ed1-ad6b-1a0e79412a79',
    //               name: 'Ответственный за отправку КП ОЭ',
    //               code: 'SENDER-KP-OE',
    //               realmId: 'PABP',
    //             },
    //             {
    //               id: 'b697c922-5a7c-4d72-ab60-2f46f6b55c3a',
    //               name: 'Ответственный за работу с карточками СНУ',
    //               code: 'SNU_RESPONSIBLE',
    //               realmId: 'PABP',
    //             },
    //             {
    //               id: '87a42af9-978f-442a-831c-be431d31ad96',
    //               name: 'Администратор ПАБП',
    //               code: 'admin',
    //               realmId: 'PABP',
    //             },
    //             {
    //               id: '79a280fe-f0a8-4bcc-b64c-71ae1a3d43c0',
    //               name: 'Базовый пользователь ПАиО',
    //               code: 'BASE_USER_PAIO',
    //               realmId: 'PABP',
    //             },
    //             {
    //               id: '3b25b467-a793-4e94-9532-4f6c877140f8',
    //               name: 'Ответственный за сбор данных',
    //               code: 'DATA_EXTRACTOR',
    //               realmId: 'PABP',
    //             },
    //             {
    //               id: '466e18dd-e459-4c76-be32-cbf0845c60c0',
    //               name: 'Ответственный ОИОЭиРН (анализ СНУ)',
    //               code: 'SNU_OIOEIRN',
    //               realmId: 'PABP',
    //             },
    //           ],
    //           subdivision: '128467',
    //           subdivisionTxt: 'Участок №1',
    //           groups: [
    //             {
    //               id: '5c0d2e87-b013-49c1-ada7-5f449566bb28',
    //               name: 'BAL',
    //             },
    //           ],
    //           email: 'admin_admin@click-group.ru',
    //           realmId: 'PABP',
    //           login: 'admin',
    //           created: '2023-02-14T12:48:02.629+00:00',
    //           password: '',
    //           active: true,
    //           isUnsi: false,
    //           position: 'Инженер',
    //           domain: {
    //             id: 1,
    //             loginPrefix: 'co',
    //             description: 'Ldap Центрального Аппарата',
    //             alias: 'Центральный аппарат',
    //             code: 'CO',
    //             federationLink: '2deebc32-7622-446a-bb0a-f4bcd26178c9',
    //           },
    //           label: 'Админов Админ Админович - Инженер - Участок №1',
    //           updatedDate: '2023-12-19T09:49:49.429+00:00',
    //         },
    //         pabpRNpp: {
    //           id: 7,
    //           createdBy: null,
    //           createdDate: '2023-03-02T13:27:25.027+00:00',
    //           updatedBy: null,
    //           updatedDate: null,
    //           shortName: 'ЛЕН',
    //           fullName: 'Ленинградская АЭС',
    //           code: 'LEN',
    //           priznak: 'AES',
    //           label: 'Ленинградская АЭС',
    //         },
    //         nonRemovable: true,
    //       },
    //       {
    //         id: '3e9647a095a',
    //         isTableInstruction: true,
    //         status: {
    //           id: 63,
    //           code: 'ASSIGN_NEW',
    //           status: 'Новый',
    //           docType: {
    //             id: 'ASSIGN',
    //             name: 'Поручение',
    //             viewLink: '/instructions/view/',
    //             dbTable: 'assign_f_doc',
    //           },
    //           label: 'Новый',
    //         },
    //         type: {
    //           id: 2,
    //           name: 'Стандартный',
    //           code: 'STANDARD',
    //         },
    //         dueNumber: 11,
    //         assignee: '014ffb3a-5bf4-48a1-8f02-64c22be0d003',
    //         approver: '014ffb3a-5bf4-48a1-8f02-64c22be0d003',
    //         body: 'Ознакомиться с результатами согласования ФПП/ФСС, оповестить ответственное за внедрение подразделение',
    //         initiator: {
    //           id: '61fa7ea7-0daa-461c-8413-cb96c77cf2d9',
    //           unsiUserId: null,
    //           firstName: 'Админ Админович',
    //           lastName: 'Админов',
    //           roles: [
    //             {
    //               id: 'f05d2f38-7e77-4ed1-ad6b-1a0e79412a79',
    //               name: 'Ответственный за отправку КП ОЭ',
    //               code: 'SENDER-KP-OE',
    //               realmId: 'PABP',
    //             },
    //             {
    //               id: 'b697c922-5a7c-4d72-ab60-2f46f6b55c3a',
    //               name: 'Ответственный за работу с карточками СНУ',
    //               code: 'SNU_RESPONSIBLE',
    //               realmId: 'PABP',
    //             },
    //             {
    //               id: '87a42af9-978f-442a-831c-be431d31ad96',
    //               name: 'Администратор ПАБП',
    //               code: 'admin',
    //               realmId: 'PABP',
    //             },
    //             {
    //               id: '79a280fe-f0a8-4bcc-b64c-71ae1a3d43c0',
    //               name: 'Базовый пользователь ПАиО',
    //               code: 'BASE_USER_PAIO',
    //               realmId: 'PABP',
    //             },
    //             {
    //               id: '3b25b467-a793-4e94-9532-4f6c877140f8',
    //               name: 'Ответственный за сбор данных',
    //               code: 'DATA_EXTRACTOR',
    //               realmId: 'PABP',
    //             },
    //             {
    //               id: '466e18dd-e459-4c76-be32-cbf0845c60c0',
    //               name: 'Ответственный ОИОЭиРН (анализ СНУ)',
    //               code: 'SNU_OIOEIRN',
    //               realmId: 'PABP',
    //             },
    //           ],
    //           subdivision: '128467',
    //           subdivisionTxt: 'Участок №1',
    //           groups: [
    //             {
    //               id: '5c0d2e87-b013-49c1-ada7-5f449566bb28',
    //               name: 'BAL',
    //             },
    //           ],
    //           email: 'admin_admin@click-group.ru',
    //           realmId: 'PABP',
    //           login: 'admin',
    //           created: '2023-02-14T12:48:02.629+00:00',
    //           password: '',
    //           active: true,
    //           isUnsi: false,
    //           position: 'Инженер',
    //           domain: {
    //             id: 1,
    //             loginPrefix: 'co',
    //             description: 'Ldap Центрального Аппарата',
    //             alias: 'Центральный аппарат',
    //             code: 'CO',
    //             federationLink: '2deebc32-7622-446a-bb0a-f4bcd26178c9',
    //           },
    //           label: 'Админов Админ Админович - Инженер - Участок №1',
    //           updatedDate: '2023-12-19T09:49:49.429+00:00',
    //         },
    //         pabpRNpp: {
    //           id: 8,
    //           createdBy: null,
    //           createdDate: '2023-03-02T13:27:25.027+00:00',
    //           updatedBy: null,
    //           updatedDate: null,
    //           shortName: 'НВО',
    //           fullName: 'Нововоронежская АЭС',
    //           code: 'NVO',
    //           priznak: 'AES',
    //           label: 'Нововоронежская АЭС',
    //         },
    //         nonRemovable: true,
    //       },
    //       {
    //         id: 'e9647a095ab',
    //         isTableInstruction: true,
    //         status: {
    //           id: 63,
    //           code: 'ASSIGN_NEW',
    //           status: 'Новый',
    //           docType: {
    //             id: 'ASSIGN',
    //             name: 'Поручение',
    //             viewLink: '/instructions/view/',
    //             dbTable: 'assign_f_doc',
    //           },
    //           label: 'Новый',
    //         },
    //         type: {
    //           id: 2,
    //           name: 'Стандартный',
    //           code: 'STANDARD',
    //         },
    //         dueNumber: 11,
    //         assignee: '56eed081-d352-410c-83a1-db2b654eb641',
    //         approver: '56eed081-d352-410c-83a1-db2b654eb641',
    //         body: 'Ознакомиться с результатами согласования ФПП/ФСС, оповестить ответственное за внедрение подразделение',
    //         initiator: {
    //           id: '61fa7ea7-0daa-461c-8413-cb96c77cf2d9',
    //           unsiUserId: null,
    //           firstName: 'Админ Админович',
    //           lastName: 'Админов',
    //           roles: [
    //             {
    //               id: 'f05d2f38-7e77-4ed1-ad6b-1a0e79412a79',
    //               name: 'Ответственный за отправку КП ОЭ',
    //               code: 'SENDER-KP-OE',
    //               realmId: 'PABP',
    //             },
    //             {
    //               id: 'b697c922-5a7c-4d72-ab60-2f46f6b55c3a',
    //               name: 'Ответственный за работу с карточками СНУ',
    //               code: 'SNU_RESPONSIBLE',
    //               realmId: 'PABP',
    //             },
    //             {
    //               id: '87a42af9-978f-442a-831c-be431d31ad96',
    //               name: 'Администратор ПАБП',
    //               code: 'admin',
    //               realmId: 'PABP',
    //             },
    //             {
    //               id: '79a280fe-f0a8-4bcc-b64c-71ae1a3d43c0',
    //               name: 'Базовый пользователь ПАиО',
    //               code: 'BASE_USER_PAIO',
    //               realmId: 'PABP',
    //             },
    //             {
    //               id: '3b25b467-a793-4e94-9532-4f6c877140f8',
    //               name: 'Ответственный за сбор данных',
    //               code: 'DATA_EXTRACTOR',
    //               realmId: 'PABP',
    //             },
    //             {
    //               id: '466e18dd-e459-4c76-be32-cbf0845c60c0',
    //               name: 'Ответственный ОИОЭиРН (анализ СНУ)',
    //               code: 'SNU_OIOEIRN',
    //               realmId: 'PABP',
    //             },
    //           ],
    //           subdivision: '128467',
    //           subdivisionTxt: 'Участок №1',
    //           groups: [
    //             {
    //               id: '5c0d2e87-b013-49c1-ada7-5f449566bb28',
    //               name: 'BAL',
    //             },
    //           ],
    //           email: 'admin_admin@click-group.ru',
    //           realmId: 'PABP',
    //           login: 'admin',
    //           created: '2023-02-14T12:48:02.629+00:00',
    //           password: '',
    //           active: true,
    //           isUnsi: false,
    //           position: 'Инженер',
    //           domain: {
    //             id: 1,
    //             loginPrefix: 'co',
    //             description: 'Ldap Центрального Аппарата',
    //             alias: 'Центральный аппарат',
    //             code: 'CO',
    //             federationLink: '2deebc32-7622-446a-bb0a-f4bcd26178c9',
    //           },
    //           label: 'Админов Админ Админович - Инженер - Участок №1',
    //           updatedDate: '2023-12-19T09:49:49.429+00:00',
    //         },
    //         pabpRNpp: {
    //           id: 2,
    //           createdBy: '61fa7ea7-0daa-461c-8413-cb96c77cf2d9',
    //           createdDate: '2023-03-02T13:27:25.027+00:00',
    //           updatedBy: '61fa7ea7-0daa-461c-8413-cb96c77cf2d9',
    //           updatedDate: '2023-05-04T11:02:34.116+00:00',
    //           shortName: 'БЕЛ',
    //           fullName: 'Белоярская АЭС',
    //           code: 'BEL',
    //           priznak: 'AES',
    //           label: 'Белоярская АЭС',
    //         },
    //         nonRemovable: true,
    //       },
    //     ],
    //     isFromPrg: null,
    //     oldNarReport: null,
    //     orderOwnerLogin: null,
    //     taskDefinition: {
    //       PP: 398,
    //     },
    //     agentList: null,
    //     isShort: null,
    //     attachedLinkTemp: [],
    //     OTKLON_EXPERT_DKBiP: null,
    //     attachedFileTemp: [],
    //     dataExtractorLogin: null,
    //     actionResultive: null,
    //     processDescr: 'ФПП/ФСС',
    //     projectRepeatedState: null,
    //     oldOtklonReport: null,
    //     snusToSave: null,
    //     orderExecutorApproverLogin: null,
    //     isShortBranch: null,
    //     CHAIRMAN_COMMISION: null,
    //     initiatorLogin: '61fa7ea7-0daa-461c-8413-cb96c77cf2d9',
    //     referenceTables: null,
    //     responsibleLogin: null,
    //     expertVniaesLogin: null,
    //     isAdditionalInvestigation: null,
    //     isFromAct: null,
    //     startDate: null,
    //   },
    // },
    // {
    //   id: 24906,
    //   taskId: '5082974',
    //   processName: 'Поручение - 5082828',
    //   taskName: 'Ознакомление с содержанием назначенного поручения',
    //   startDate: 1718699648218,
    //   dueDate: 1718872447000,
    //   formKey: 'IntroductionСontentAssignment',
    //   localVariables: {
    //     equipmentOwnerLogin: null,
    //     agent: null,
    //     addReportReasons: [
    //       {
    //         id: 0,
    //         code: 'CCERCV',
    //         name: 'Полнота и корректность установления коренных причин нарушений',
    //       },
    //     ],
    //     actionFDocTemp: null,
    //     isPates: false,
    //     assignFDocTemp: null,
    //     dueDate: '2024-06-20T11:34:07',
    //     organizationAddReport: {
    //       id: 0,
    //       level: 'RTN',
    //       name: 'РТН',
    //     },
    //     commentHistory: [],
    //     workerGroupLogin: null,
    //     orderExecutorLogin: '61fa7ea7-0daa-461c-8413-cb96c77cf2d9',
    //     instructionList: [
    //       {
    //         assign: {
    //           id: 1209,
    //           createdBy: '014ffb3a-5bf4-48a1-8f02-64c22be0d003',
    //           createdDate: '2024-06-18T08:33:51.582+00:00',
    //           updatedBy: '014ffb3a-5bf4-48a1-8f02-64c22be0d003',
    //           updatedDate: '2024-06-18T08:33:51.595+00:00',
    //           status: {
    //             id: 63,
    //           },
    //           body: 'Определение состава комиссии с назначением ответственного за осуществление контроля качества расследования. Составление и утверждение организационно-распорядительного документа о формировании комиссии и проведении расследования события на АС',
    //           dtCreate: null,
    //           planedDateComplete: '2024-06-20T11:34:07',
    //           factDateComplete: null,
    //           documentOsnLink: '/deviationsPlanShedules/view/133',
    //           documentOsnNumber: 'БАЛ-31',
    //           initiator: '014ffb3a-5bf4-48a1-8f02-64c22be0d003',
    //           assignee: '61fa7ea7-0daa-461c-8413-cb96c77cf2d9',
    //           approver: '61fa7ea7-0daa-461c-8413-cb96c77cf2d9',
    //           completionReport: null,
    //           camundaProcId: null,
    //           type: {
    //             id: 3,
    //             name: 'Особый',
    //             code: 'SPECIAL',
    //           },
    //           number: 140,
    //           pabpRNpp: {
    //             id: 1,
    //             createdBy: '61fa7ea7-0daa-461c-8413-cb96c77cf2d9',
    //             createdDate: '2023-03-02T13:27:25.027+00:00',
    //             updatedBy: '61fa7ea7-0daa-461c-8413-cb96c77cf2d9',
    //             updatedDate: '2023-05-04T10:55:28.110+00:00',
    //             shortName: 'БАЛ',
    //             fullName: 'Балаковская АЭС',
    //             code: 'BAL',
    //             priznak: 'AES',
    //             label: 'Балаковская АЭС',
    //           },
    //           assigneeContr: null,
    //           numberString: 'БАЛ-140',
    //           dueNumber: 2,
    //           renewalDate: null,
    //           nameForFrontend: '1209',
    //           dueDate: '2024-06-20T11:34:07',
    //         },
    //       },
    //     ],
    //     isFromPrg: null,
    //     oldNarReport: 10,
    //     orderOwnerLogin: null,
    //     taskDefinition: {
    //       ASSIGN: 1209,
    //     },
    //     agentList: [],
    //     isShort: null,
    //     attachedLinkTemp: null,
    //     OTKLON_EXPERT_DKBiP: null,
    //     attachedFileTemp: null,
    //     dataExtractorLogin: '014ffb3a-5bf4-48a1-8f02-64c22be0d003',
    //     actionResultive: null,
    //     processDescr: 'Поручение',
    //     projectRepeatedState: null,
    //     oldOtklonReport: null,
    //     snusToSave: null,
    //     orderExecutorApproverLogin: '61fa7ea7-0daa-461c-8413-cb96c77cf2d9',
    //     isShortBranch: false,
    //     CHAIRMAN_COMMISION: '014ffb3a-5bf4-48a1-8f02-64c22be0d003',
    //     initiatorLogin: '014ffb3a-5bf4-48a1-8f02-64c22be0d003',
    //     referenceTables: {
    //       nar_r_subsys: {
    //         key: 'narRPnagId',
    //         pnag: 4,
    //       },
    //       nar_r_category: {
    //         key: 'pnagId',
    //         pnag: 4,
    //       },
    //       nar_r_cs_r: {
    //         key: 'narRPnagId',
    //         pnag: 4,
    //       },
    //       nar_r_sys_dvc: {
    //         key: 'narRPnagId',
    //         pnag: 4,
    //       },
    //       nar_r_cs_d: {
    //         key: 'narRPnagId',
    //         pnag: 4,
    //       },
    //       nar_r_feature: {
    //         key: 'narRPnagId',
    //         pnag: 4,
    //       },
    //       nar_r_impact: {
    //         key: 'narRPnagId',
    //         pnag: 4,
    //       },
    //       nar_r_sys: {
    //         key: 'narRPnagId',
    //         pnag: 4,
    //       },
    //     },
    //     responsibleLogin: null,
    //     expertVniaesLogin: null,
    //     isAdditionalInvestigation: true,
    //     isFromAct: null,
    //     startDate: null,
    //   },
    // },
  ],
  totalElements: 108,
  totalPages: 11,
};
