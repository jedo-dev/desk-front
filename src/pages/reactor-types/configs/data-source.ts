import { TableProps } from 'antd';

export const reactorTypesDataSource: TableProps['dataSource'] = [
  {
    id: 'test_kol',
    code_name: 'ВВЭР',
    full_name: 'Водно-водяной энергетический реактор',
    description: 'Водно-водяной энергетический реактор',
    is_active: true,
  },
  {
    id: 'test_bel',
    code_name: 'РБМК',
    full_name: 'Реактор большой мощности канальный',
    description: 'Реактор большой мощности канальный',
    is_active: true,
  },
  {
    id: 'test_kln',
    code_name: 'БН',
    full_name: 'Реактор на быстрых нейтронах',
    description: 'Реактор на быстрых нейтронах',
    is_active: false,
  },
];
