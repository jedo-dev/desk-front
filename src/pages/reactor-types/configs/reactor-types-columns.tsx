import { Checkbox, TableProps } from 'antd';

import { EditButtonRender } from '../lib/EditButtonRender';

export const reactorTypesColumns: TableProps['columns'] = [
  {
    key: 'code_name',
    title: 'Код реактора',
    dataIndex: 'code_name',
    align: 'center',
    filters: [
      {
        text: 'КОЛ АЭС',
        value: 'КОЛ АЭС',
      },
      {
        text: 'БЕЛ АЭС',
        value: 'БЕЛ АЭС',
      },
    ],
    onFilter: (value, record) => record.code_name.indexOf(value) === 0,
  },
  {
    key: 'full_name',
    title: 'Наименование',
    dataIndex: 'full_name',
    align: 'center',
    filters: [
      {
        text: 'Кольская АЭС',
        value: 'Кольская АЭС',
      },
      {
        text: 'Белоярская АЭС',
        value: 'Белоярская АЭС',
      },
    ],
    onFilter: (value, record) => record.code_name.indexOf(value) === 0,
    sorter: (a, b) => a.full_name.length - b.full_name.length,
  },
  {
    key: 'description',
    title: 'Описание',
    dataIndex: 'description',
    align: 'center',
    sorter: (a, b) => a.description.length - b.description.length,
  },
  {
    key: 'is_active',
    title: 'Активный',
    dataIndex: 'is_active',
    align: 'center',
    render: (_, record) => {
      return <Checkbox defaultChecked={record.is_active} disabled></Checkbox>;
    },
  },
  {
    key: 'actions',
    title: 'Действия',
    dataIndex: 'id',
    align: 'center',
    render: (_, record) => <EditButtonRender record={record} />,
  },
];
