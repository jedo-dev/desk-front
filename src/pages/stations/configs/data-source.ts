import { TableProps } from 'antd';
import { IStations } from '../../../entities/stations/model/stations.model';

export const stationsDataSource: TableProps['dataSource'] & IStations[] = [
  {
    id: 'test_kol',
    code_name: 'КОЛ АЭС',
    full_name: 'Кольская АЭС',
    description: 'Кольская атомная электростанция',
    is_active: true,
  },
  {
    id: 'test_bel',
    code_name: 'БЕЛ АЭС',
    full_name: 'Белоярская АЭС',
    description: 'АО “Концерн Росэнергоатом”',
    is_active: true,
  },
  {
    id: 'test_kln',
    code_name: 'КЛН АЭС',
    full_name: 'Калининская АЭС',
    description: 'Калининская атомная электростанция',
    is_active: true,
  },
  {
    id: 'test_rst',
    code_name: 'РСТ АЭС',
    full_name: 'Ростовская АЭС',
    description: 'Ростовская атомная электростанция',
    is_active: true,
  },
  {
    id: 'test_smo',
    code_name: 'СМО АЭС',
    full_name: 'Смоленская АЭС',
    description: 'Смоленская атомная электростанция',
    is_active: false,
  },
];
