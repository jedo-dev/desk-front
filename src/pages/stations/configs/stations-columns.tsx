import { TableProps, Checkbox } from 'antd';

import { EditButtonRender } from '../lib/EditButtonRender';
import { SearchOutlined } from '@ant-design/icons';

export const stationsColumns: TableProps['columns'] = [
  {
    key: 'code_name',
    title: 'Код станции',
    dataIndex: 'code_name',
    align: 'center',
    filters: [],
    filterIcon: (filtered: boolean) => (
      <SearchOutlined style={{ color: filtered ? '#1677ff' : undefined }} />
    ),
  },
  {
    key: 'full_name',
    title: 'Название станции',
    dataIndex: 'full_name',
    align: 'center',
    filters: [],
    filterIcon: (filtered: boolean) => (
      <SearchOutlined style={{ color: filtered ? '#1677ff' : undefined }} />
    ),
  },
  {
    key: 'description',
    title: 'Описание',
    dataIndex: 'description',
    align: 'center',
    filters: [],
    filterIcon: (filtered: boolean) => (
      <SearchOutlined style={{ color: filtered ? '#1677ff' : undefined }} />
    ),
  },
  {
    key: 'is_active',
    title: 'Активный',
    dataIndex: 'is_active',
    align: 'center',
    filters: [
      {
        text: 'Да',
        value: 'Да',
      },
      {
        text: 'Нет',
        value: 'Нет',
      },
    ],
    render: (_, record) => {
      return <Checkbox defaultChecked={record.is_active} disabled></Checkbox>;
    },
  },
  {
    key: 'actions',
    title: 'Действия',
    dataIndex: 'id',
    align: 'center',
    render: (_, record) => <EditButtonRender record={record} />,
  },
];
