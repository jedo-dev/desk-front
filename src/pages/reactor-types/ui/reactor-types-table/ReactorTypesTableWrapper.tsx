import { ReusedTable } from '@shared/components';
import { TableProps } from 'antd';
import { reactorTypesColumns } from '../../configs/reactor-types-columns';

export const ReactorTypesTableWrapper: React.FC<{
  dataSource: TableProps['dataSource'];
}> = (props) => {
  return (
    <ReusedTable
      keyTable='eventsTable'
      rowKey='id'
      columns={reactorTypesColumns}
      dataSource={props.dataSource}
      scroll={{ x: 'max-content' }}
    />
  );
};
