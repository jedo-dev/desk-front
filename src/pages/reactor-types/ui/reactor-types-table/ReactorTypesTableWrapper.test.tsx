import { withReduxProvider } from '@app/providers/redux';
import { matchMediaConfig } from '@shared/test-configs/window-match-media';
import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import { TableProps } from 'antd';
import { describe, expect, it } from 'vitest';
import { reactorTypesColumns } from '../../configs/reactor-types-columns';
import { ReactorTypesTableWrapper } from './ReactorTypesTableWrapper';

matchMediaConfig;

//мок данные таблицы
const mockDataSource: TableProps['dataSource'] = [
  {
    code_name: 'КОЛ АЭС',
  },
];

describe('StationsTable', () => {
  it('Конфигурация колонок соответсвует макету для компонента StationsTable', () => {
    expect(reactorTypesColumns).toMatchSnapshot();
  });

  it('Таблица должна отрендериться с валидными данными', () => {
    render(withReduxProvider(() => <ReactorTypesTableWrapper dataSource={mockDataSource} />)());
    expect(screen.getByText('Код реактора')).toBeInTheDocument();
  });

  it('Таблица должна отрендериться с пустыми данными', () => {
    render(withReduxProvider(() => <ReactorTypesTableWrapper dataSource={[]} />)());
    expect(screen.getByText('Код реактора')).toBeInTheDocument();
  });

  it('Таблица должна отрендериться с undefined данными', () => {
    render(withReduxProvider(() => <ReactorTypesTableWrapper dataSource={undefined} />)());
    expect(screen.getByText('Код реактора')).toBeInTheDocument();
  });
});
