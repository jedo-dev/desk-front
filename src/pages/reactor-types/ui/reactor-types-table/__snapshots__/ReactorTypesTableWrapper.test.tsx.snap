// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`StationsTable > Конфигурация колонок соответсвует макету для компонента StationsTable 1`] = `
[
  {
    "align": "center",
    "dataIndex": "code_name",
    "filters": [
      {
        "text": "КОЛ АЭС",
        "value": "КОЛ АЭС",
      },
      {
        "text": "БЕЛ АЭС",
        "value": "БЕЛ АЭС",
      },
    ],
    "key": "code_name",
    "onFilter": [Function],
    "title": "Код реактора",
  },
  {
    "align": "center",
    "dataIndex": "full_name",
    "filters": [
      {
        "text": "Кольская АЭС",
        "value": "Кольская АЭС",
      },
      {
        "text": "Белоярская АЭС",
        "value": "Белоярская АЭС",
      },
    ],
    "key": "full_name",
    "onFilter": [Function],
    "sorter": [Function],
    "title": "Наименование",
  },
  {
    "align": "center",
    "dataIndex": "description",
    "key": "description",
    "sorter": [Function],
    "title": "Описание",
  },
  {
    "align": "center",
    "dataIndex": "is_active",
    "key": "is_active",
    "render": [Function],
    "title": "Активный",
  },
  {
    "align": "center",
    "dataIndex": "id",
    "key": "actions",
    "render": [Function],
    "title": "Действия",
  },
]
`;
